### PR TITLE
Fri 496

### PR DESCRIPTION
**Pull request by QuyenLy87**
This pull request is for https://jira.ihtsdotools.org/browse/FRI-496
A new category called "delta-file-required" will be added to the assertions that require to verify the Delta content
If a release does not contain the Delta content, then those assertions will not be triggered.

### CODE REVIEW

File: execution-service/src/main/java/org/ihtsdo/rvf/execution/service/MysqlValidationService.java [hunk 1]
```diff
@@ -165,6 +171,13 @@ private void runAssertionTests( ValidationStatusReport statusReport, MysqlExecut
 		for (final AssertionGroup group : groups) {
 			assertions.addAll(group.getAssertions());
 		}
+
+		// filter the delta-file-required assertions out if the release does not contain the delta files
+		if (validationConfig.isMissingRf2DeltaFromRelease()) {
```
Comment (CoderMChu) [inline execution-service/src/main/java/org/ihtsdo/rvf/execution/service/MysqlValidationService.java:176]: @QuyenLy87 We could change the parameter to skipValidationForDeltaFiles

File: execution-service/src/main/java/org/ihtsdo/rvf/execution/service/ValidationVersionLoader.java [hunk 2]
```diff
@@ -119,8 +119,7 @@ public void loadProspectiveVersion(ValidationStatusReport statusReport, MysqlExe
 		}
 
 		if (!validationConfig.isRf2DeltaOnly() && !checkDeltaFilesExist(validationConfig.getLocalProspectiveFile())) {
-			final String schemaName = prospectiveVersion.startsWith(RVF_DB_PREFIX) ? prospectiveVersion : RVF_DB_PREFIX + prospectiveVersion;
-			releaseDataManager.insertIntoProspectiveDeltaTables(schemaName, executionConfig);
+			validationConfig.setMissingRf2DeltaFromRelease(true);
```
Comment (CoderMChu) [inline execution-service/src/main/java/org/ihtsdo/rvf/execution/service/ValidationVersionLoader.java:122]: @QuyenLy87  I think there are other assertions not for "release-type-validation" also require delta. e.g component-centric validations for current authoring cycle only. If the delta is empty in the mysql db then these assertions will always pass which is bad.

File: importer/src/main/java/org/ihtsdo/rvf/importer/AssertionGroupImporter.java [hunk 3]
```diff
@@ -348,6 +348,7 @@ public boolean isImportRequired() {
 	 * file-centric-validation
 	 * release-type-validation
 	 * component-centric-validation
+	 * delta-file-required
```
Comment (CoderMChu) [inline importer/src/main/java/org/ihtsdo/rvf/importer/AssertionGroupImporter.java:356]: @QuyenLy87 As discussed in the call, we don't need to create a new assertion group for delta-file-required.

**Review by CoderMChu — COMMENTED** [review]
@QuyenLy87 This turns out to be complicated as existing assertions were written based on release package having delta, snapshot and full files. However since FRI the international release is no longer having delta files. Only daily builds still have delta files. We could update existing category for release-type-validation to include the file type. e.g release-type-validation-delta, release-type-validation-snapshot and release-type-validation-full. Let me know what you think of this. Thanks.

File: execution-service/src/main/java/org/ihtsdo/rvf/execution/service/MysqlValidationService.java (hunk 1 above)
Comment (QuyenLy87) [reply execution-service/src/main/java/org/ihtsdo/rvf/execution/service/MysqlValidationService.java]: Done

File: importer/src/main/java/org/ihtsdo/rvf/importer/AssertionGroupImporter.java (hunk 3 above)
Comment (QuyenLy87) [reply importer/src/main/java/org/ihtsdo/rvf/importer/AssertionGroupImporter.java]: Done

**Pull request closed without merging**